### PR TITLE
Doc for mystery number

### LINKED ISSUE #66: Documentation for mystery number
## 📚 Documentation

Documentation for mystery number.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

Please assign this to me under GSSOC'21.

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Doc for mystery number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #66

### CODE REVIEW

**Review by HarshCasper — APPROVED** [review]

File: docs/mathematics/mystery-number.mdx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+Clearly, the number 122 and 221 are reverse of each other.
+
+So, X = 343 is a Mystery number.
+```
+
+## Algorithm
+
+  - Given a number `n` to check if it is a mystery number or not.
+  - Iterate through every possible pair `(n1, n2)`, where `n1` and `n2` are reverse of each other and the sum, `n1 + n2` is smaller than or equal to `n`.
+  - If we get a pair `(n1, n2)`, such that `n1 + n2 = n` and `n1` and `n2` are reverse of each other, then we print the pair `(n1, n2)` and return `TRUE`.
```
Comment (rohanbabbar04) [inline docs/mathematics/mystery-number.mdx:28]: Write true not TRUE

File: docs/mathematics/mystery-number.mdx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  - Else, we return `FALSE`.
```
Comment (rohanbabbar04) [inline docs/mathematics/mystery-number.mdx:29]: Write false not FALSE

**Review by rohanbabbar04 — CHANGES_REQUESTED** [review]
Also as per the java code it returns an integer instead of a boolean value so it would be good if you can change it to boolean.

**Comment by nidh-ish** [comment]
> Also as per the java code it returns an integer instead of a boolean value so it would be good if you can change it to boolean.

Instead of that, in the Algorithm part, I can write return true/1 and false/-1. It would be better, I guess. What do you say?

**Comment by nidh-ish** [comment]
@rohanbabbar04 I have made it boolean.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**